plugins/table/plugin.js

```diff
@@ -30,7 +30,22 @@ CKEDITOR.plugins.add( 'table', {
 					element: 'table',
 					right: function( element ) {
 						if ( element.styles ) {
-							if ( element.styles.border && element.styles.border.match( /solid/ ) ) {
+							var parsedStyle;
+							if ( element.styles.border ) {
+								parsedStyle = CKEDITOR.tools.style.parse.border( element.styles.border );
+							} else if ( CKEDITOR.env.ie && CKEDITOR.env.version === 8 ) {
+								var styleData = element.styles;
+								// Workaround for IE8 browser. IE8 after parsing CSS border param changes it for it
+								// basic comoponents(border-top...border-right). We have to check if those parameters exists and
+								// have the same value to get expected editor behavior. (#566)
+								if ( styleData[ 'border-left' ] && styleData[ 'border-left' ] === styleData[ 'border-right' ] &&
+										styleData[ 'border-right' ] === styleData[ 'border-top' ] &&
+										styleData[ 'border-top' ] === styleData[ 'border-bottom' ] ) {
```

Comment:
These two lines (42 and 43) are not properly indented, should be one `tab` less:
```
if ( styleData[ 'border-left' ] && styleData[ 'border-left' ] === styleData[ 'border-right' ] &&
    styleData[...
```

you may put one empty line before `if` body to increase readability.